FirebaseVertexAI/Sample/ImagenScreen/ImagenViewModel.swift

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseVertexAI
+import Foundation
+import OSLog
+import SwiftUI
+
+@MainActor
+class ImagenViewModel: ObservableObject {
+  private var logger = Logger(subsystem: Bundle.main.bundleIdentifier!, category: "generative-ai")
+
+  @Published
+  var userInput: String = ""
+
+  @Published
+  var images = [UIImage]()
+
+  @Published
+  var errorMessage: String?
+
+  @Published
+  var inProgress = false
+
+  private let model: ImagenModel
+
+  // 1. Initialize the Vertex AI service
+  private let vertexAI = VertexAI.vertexAI()
+
+  init() {
+    // 2. Configure Imagen settings
+    let modelName = "imagen-3.0-generate-002"
+    let safetySettings = ImagenSafetySettings(
+      safetyFilterLevel: .blockLowAndAbove
+    )
+    var generationConfig = ImagenGenerationConfig()
+    generationConfig.numberOfImages = 4
+    generationConfig.aspectRatio = .landscape4x3
+    generationConfig.addWatermark = false
```

Comment:
nit: I'd recommend removing `addWatermark` for future Developer API support where all images get a watermark (`true` also throws an error there).